pkg/network/dns/cache.go

```diff
@@ -3,30 +3,38 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-//go:build windows || linux_bpf
-// +build windows linux_bpf
+//go:build (windows && npm) || linux_bpf
+// +build windows,npm linux_bpf
 
 package dns
 
 import (
 	"sync"
 	"time"
 
-	"go.uber.org/atomic"
-
+	nettelemetry "github.com/DataDog/datadog-agent/pkg/network/telemetry"
 	"github.com/DataDog/datadog-agent/pkg/process/util"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-type reverseDNSCache struct {
-	// Telemetry
-	length    *atomic.Int64
-	lookups   *atomic.Int64
-	resolved  *atomic.Int64
-	added     *atomic.Int64
-	expired   *atomic.Int64
-	oversized *atomic.Int64
+// Telemetry
+var cacheTelemetry = struct {
+	length    nettelemetry.StatGaugeWrapper
+	lookups   nettelemetry.StatGaugeWrapper
+	resolved  nettelemetry.StatGaugeWrapper
+	added     nettelemetry.StatGaugeWrapper
+	expired   nettelemetry.StatGaugeWrapper
+	oversized nettelemetry.StatGaugeWrapper
+}{
+	nettelemetry.NewStatGaugeWrapper(dnsModuleName, "length", []string{}, "Gauge measuring the current size of the DNS cache"),
+	nettelemetry.NewStatGaugeWrapper(dnsModuleName, "lookups", []string{}, "Gauge measuring the number of lookups to the DNS cache"),
+	nettelemetry.NewStatGaugeWrapper(dnsModuleName, "resolved", []string{}, "Gauge measuring the number of successful lookups to the DNS cache"),
+	nettelemetry.NewStatGaugeWrapper(dnsModuleName, "added", []string{}, "Gauge measuring the number of additions to the DNS cache"),
+	nettelemetry.NewStatGaugeWrapper(dnsModuleName, "expired", []string{}, "Gauge measuring the number of failed lookups to the DNS cache"),
+	nettelemetry.NewStatGaugeWrapper(dnsModuleName, "oversized", []string{}, "Gauge measuring the number of lookups to the DNS cache that reached the max domains per IP limit"),
```

Comment:
I think everything but `length` is better described by a counter. WDYT? Maybe you can add a new wrapper for counters? I think you can use `Delete()` to reset the value for test purposes.

Reply:
added new counter wrapper, and migrated any monotonically increasing metric to counter